### PR TITLE
Events and conditionals

### PR DESCRIPTION
**Pull request by aszlam**
# Description

This adds support for events and conditionals to interpreter, and updates DSL accordingly.  control flow interpretation is taken out of action handlers and put into a general event interpreter.

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ x Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

### CODE REVIEW

**Comment by aszlam** [comment]
This is ready for review now.  Still need to do the locobot/robot code, and we need to decide what to do about "undo"

**Comment by kavyasrinet** [comment]
> This is ready for review now. Still need to do the locobot/robot code, and we need to decide what to do about "undo"

I vote for skipping "undo" for now and revisit, haven't seen much in human interaction data.

File: droidlet/task/task.py [hunk 1]
```diff
@@ -235,26 +184,97 @@ class ControlBlock(Task):
...
+            # check if we have reached the end of our task list, go back to beginning
+            if self.task_list_idx == len(self.task_fns):
+                self.task_list_idx = 0
+                self.run_count += 1
+                self_mem.update_task(task=self)
+            # can only be here if
+            # there is no child, so previous generated child task is finished.
+            # start the next child in the sequence:
+            g = self.task_fns[self.task_list_idx]
+            # if g is a Task that has been wrapped via task_to_generator, light its fuse
```
Comment (kavyasrinet) [inline droidlet/task/task.py:263]: light its fuse -> "activate it" for clarity ? :)

**Review by kavyasrinet — COMMENTED** [review]
Look good so far, we need to add tests to test these pathways

**Review by kavyasrinet — APPROVED** [review]
Let's merge for now and then fix if anything breaks

**Review by snyxan — APPROVED** [review]
Reviewed at high level except for json schema part.

**Pull request merged**